apps/src/codegen.js

```diff
@@ -468,12 +468,6 @@ exports.isNextStepSafeWhileUnwinding = function (interpreter) {
   if (state.done) {
     return true;
   }
-  if (type === "ForStatement") {
-    var mode = state.mode || 0;
-    // Safe to skip over ForStatement's in mode 0 (init) and 3 (update),
-    // but not mode 1 (test) or mode 2 (body) while unwinding...
-    return mode === 0 || mode === 3;
-  }
```

Comment:
I'm not completely sure what the impact of this part of the change will be. We will certain stop more quickly in some situations. It *could* help us avoid bugs where we step too far. It might also re-introduce a bug where we stepped in same place twice.